common/lib/api-client/request-headers.test.js

```diff
@@ -43,34 +43,38 @@ describe('API Client', function () {
         it('should return the `Idempotency-Key` header', function () {
           expect(headers['Idempotency-Key']).to.equal('#uuid')
         })
+
+        it('should return the `X-Transaction-Id` header', function () {
+          expect(headers['X-Transaction-Id']).to.equal('auto-#uuid')
+        })
       })
     })
 
     context('when called with a format', function () {
       let headers
 
       beforeEach(function () {
-        headers = getRequestHeaders('format/foo')
+        headers = getRequestHeaders({ format: 'format/foo' })
       })
 
       it('should return `Accept` header with specified format and the api version', function () {
         expect(headers.Accept).to.equal(
           `format/foo; version=${config.API.VERSION}`
         )
       })
+    })
 
-      it('should return the `User-Agent` header', function () {
-        expect(headers['User-Agent']).to.equal(
-          'hmpps-book-secure-move-frontend/chuck-d'
-        )
-      })
+    context('when called with a request', function () {
+      let headers
 
-      it('should return the `Accept-Encoding` header', function () {
-        expect(headers['Accept-Encoding']).to.equal('gzip')
+      beforeEach(function () {
+        headers = getRequestHeaders({
+          req: { transactionId: '#transactionId' },
+        })
       })
 
-      it('should return the `Idempotency-Key` header', function () {
-        expect(headers['Idempotency-Key']).to.equal('#uuid')
+      it('should return req’s transactionId as the `X-Transaction-Id` header', function () {
```

Comment:
Are we ok with removing the `Idempotency-Key` test here?

Reply:
That looks like a straight up mistake to me. Good catch

Reply:
Aah. It's not really removed - it's been moved to line 43 in the previous context block. I'll add in the dupe test though.